security-oauth2/src/main/java/io/micronaut/security/oauth2/configuration/OauthClientConfigurationProperties.java

```diff
@@ -277,6 +280,8 @@ public void setRevocation(RevocationEndpointConfigurationProperties revocation)
 
         private HeaderTokenPropagatorConfigurationProperties headerPropagation;
 
+        private Map<String, String> additonalRequestParams = Collections.emptyMap();
```

Comment:
This probably needs to be a `Map<String, Object> `to allow more flexibility.

Reply:
Just curious as to the need for flexibility here?, as it feeds directly into the `SecureGrantMap` which is typed as `Map<String, String>`

Reply:
you are correct. Let's keep a `Map<String, String>` as you suggest.